File: Assets/PythonMCP/modules/rename_object_module.py
import requests
import json
from typing import Dict


class RenameObjectModule:
    def __init__(self, base_url: str):
        self.base_url = base_url

    def execute(self, object_path: str, new_name: str) -> Dict:
        """Переименовывает объект в сцене"""
        try:
            if not object_path or not new_name:
                return {
                    "success": False,
                    "action": "rename_object",
                    "error": "object_path and new_name are required",
                }

            data = {"path": object_path, "newName": new_name}
            response = requests.put(f"{self.base_url}/objects/rename", json=data)
            response.raise_for_status()
            result = response.json()

            return {
                "success": result.get("success", False),
                "action": "rename_object",
                "data": result if result.get("success") else None,
                "error": result.get("error"),
            }
        except requests.exceptions.RequestException as e:
            return {
                "success": False,
                "action": "rename_object",
                "error": f"Request error: {str(e)}",
            }
        except json.JSONDecodeError as e:
            return {
                "success": False,
                "action": "rename_object",
                "error": f"JSON decode error: {str(e)}",
            }

File: Assets/Editor/SceneAPI/Modules/Objects/RenameObjectModule.cs
using System;
using System.Net;
using Newtonsoft.Json;
using UnityEngine;

namespace SceneAPI.Modules
{
    public static class RenameObjectModule
    {
        public static string Execute(HttpListenerContext context)
        {
            try
            {
                string requestBody = GetRequestBody(context);
                var data = JsonConvert.DeserializeObject<dynamic>(requestBody);

                string objectPath = data?.path;
                string newName = data?.newName;

                if (string.IsNullOrEmpty(objectPath) || string.IsNullOrEmpty(newName))
                {
                    return JsonConvert.SerializeObject(new
                    {
                        success = false,
                        error = "Object path and new name are required"
                    });
                }

                GameObject obj = GameObjectUtilities.FindGameObjectByPath(objectPath);
                if (obj == null)
                {
                    return JsonConvert.SerializeObject(new
                    {
                        success = false,
                        error = "Object not found"
                    });
                }

                string oldName = obj.name;
                obj.name = newName;

                // Вычисляем новый путь
                string newPath = objectPath;
                if (objectPath.Contains("/"))
                {
                    string parentPath = objectPath.Substring(0, objectPath.LastIndexOf('/'));
                    newPath = $"{parentPath}/{newName}";
                }
                else
                {
                    newPath = newName;
                }

                return JsonConvert.SerializeObject(new
                {
                    success = true,
                    oldPath = objectPath,
                    newPath = newPath,
                    oldName = oldName,
                    newName = newName,
                    message = $"Object renamed from '{oldName}' to '{newName}'"
                });
            }
            catch (Exception ex)
            {
                return JsonConvert.SerializeObject(new
                {
                    success = false,
                    error = $"Error renaming object: {ex.Message}"
                });
            }
        }

        private static string GetRequestBody(HttpListenerContext context)
        {
            using (var reader = new System.IO.StreamReader(context.Request.InputStream))
            {
                return reader.ReadToEnd();
            }
        }
    }
}